### PR TITLE
Add error details to RunFromPackage failure log

### LINKED ISSUE #8922: Include failure details in log for Run From Package startup failures
When an app using "run from package" fails due to problems downloading the zip package, the platform writes a file "FAILED TO INITIALIZE RUN FROM PACKAGE.txt", which the host looks for, and when found will refuse to start (code [here](https://github.com/Azure/azure-functions-host/blob/a68b37b1d667e5d9a3152da935c3b80384da51da/src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs#L203)). The content of this log file actually includes reasons for the failure. We should log those details as well to aid in issue diagnosis.

We should also confirm that this error message gets written to the user's Application Insights (if it isn't already) to reduce support cases.

### PR DESCRIPTION
**Pull request by madelinegordon**
### Issue describing the changes in this PR

resolves #8922 

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Added file contents to RunFromPackage failure log.

### CODE REVIEW

File: src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs [hunk 1]
```diff
@@ -200,9 +200,11 @@ private void CheckFileSystem()
                 // Check for marker file indicating a zip package failure, and if found stop the application.
                 // We never want to run with an incorrect file system.
                 string path = Path.Combine(_applicationHostOptions.CurrentValue.ScriptPath, ScriptConstants.RunFromPackageFailedFileName);
+
                 if (File.Exists(path))
                 {
-                    _logger.LogError($"Shutting down host due to presence of {path}");
+                    string errorMessage = File.ReadAllText(path);
+                    _logger.LogError($"Shutting down host due to presence of {path}. Error details: {errorMessage}");
```
Comment (kshyju) [inline src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs]: Should we use "File content" instead of Error details to be more explicit ?

**Review by kshyju — APPROVED** [review]

File: src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs [hunk 2]
```diff
@@ -200,9 +200,11 @@ private void CheckFileSystem()
                 // Check for marker file indicating a zip package failure, and if found stop the application.
                 // We never want to run with an incorrect file system.
                 string path = Path.Combine(_applicationHostOptions.CurrentValue.ScriptPath, ScriptConstants.RunFromPackageFailedFileName);
+
                 if (File.Exists(path))
                 {
-                    _logger.LogError($"Shutting down host due to presence of {path}");
+                    string errorMessage = File.ReadAllText(path);
```
Comment (kshyju) [inline src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs]: What happens if the `ReadAllText` throws any of the documented exception here? Do we have an outer catch which logs that exception?

File: src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs [hunk 3]
```diff
@@ -200,9 +200,11 @@ private void CheckFileSystem()
...
+                    _logger.LogError($"Shutting down host due to presence of {path}. Error details: {errorMessage}");
                     _applicationLifetime.StopApplication();
                 }
```
Comment (kshyju) [inline src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs:222]: If ZipDeploymentAppSettingsExist is truthy and this file does not exist, should we log a warning?

Comment (liliankasem) [reply src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs:222]: yes i think we should

**Review by liliankasem — APPROVED** [review]

File: src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs (hunk 3 above)
Comment (madelinegordon) [reply src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs:222]: From my understanding, `ZipDeploymentAppSettingsExist = true` and this file not existing is the expected behavior/happy path

Comment (madelinegordon) [reply src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs:222]: And we would only stop the host if `FAILED TO INITIALIZE RUN FROM PACKAGE.txt` exists.

**Review by mathewc — APPROVED** [review]

**Pull request merged**